bson/unmarshaling_cases_test.go

```diff
@@ -9,72 +9,164 @@ package bson
 import (
 	"reflect"
 
-	"go.mongodb.org/mongo-driver/bson/bsoncodec"
+	"go.mongodb.org/mongo-driver/bson/bsonrw"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
 )
 
 type unmarshalingTestCase struct {
 	name  string
-	reg   *bsoncodec.Registry
 	sType reflect.Type
 	want  interface{}
 	data  []byte
 }
 
-var unmarshalingTestCases = []unmarshalingTestCase{
+var unmarshalingTestCases = func() []unmarshalingTestCase {
```

Comment:
Maybe `createUnmarshalingTestCases` since it's a function now? Great new tests, though. Thanks for adding.

Reply:
Seems reasonable, will update.